pixel/head.html

```diff
@@ -1,78 +1,204 @@
+      var departments = webWidget.chat.departments;
+      var color = webWidget.color;
+      var enabled = parseValuesToInt(departments.enabled);
+      var select = isNaN(parseInt(departments.select))
+        ? departments.select
+        : parseInt(departments.select);
+      var tags = departments.tags;
+
+      window.zESettings.webWidget = {
+        color: {
+          theme: color.theme,
+        },
+        helpCenter: {
+          suppress: helpCenterSuppress,
+        },
+        chat: {
+          departments: {
+            enabled: enabled,
+            select: select,
+            tags: tags,
+          },
+        },
+      };
     }
-})
+
+    function addZDSnippet() {
+      var script = document.createElement("script");
+
+      btn.disabled = true;
+
+      script.id = "ze-snippet";
+      script.async = "async";
+      script.src =
+        "https://static.zdassets.com/ekr/snippet.js?key={{settings.accountKey}}";
+      script.onload = function () {
+        var intervalId = setInterval(function () {
+          if ("zE" in window && "activate" in zE) {
+            clearInterval(intervalId);
+
+            zE("webWidget:on", "chat:end", function () {
+              localStorage.setItem("zdChatOpen", false);
+            });
+
+            configureSnippetForVTEX();
+
+            zE.activate();
+
+            document.body.removeChild(btn);
+          }
+        }, 400);
+      };
+
+      document.head.appendChild(script);
+    }
+
+    function shouldOpenChat() {
+      var zdStore = JSON.parse(localStorage.getItem("ZD-store"));
+
+      if (zdStore) {
+        return zdStore.widgetShown;
+      }
+
+      return localStorage.getItem("zdChatOpen") === "true";
+    }
+
+    window.addEventListener("DOMContentLoaded", function () {
+      if (shouldOpenChat()) {
+        addZDSnippet();
+      } else {
+        btn.id = "zendesk-fake-btn";
+        btn.innerHTML =
+          '<svg id="zendesk-fake-icon" x="0" y="0" viewBox="0 0 15 16" xml:space="preserve" aria-hidden="true"><path d="M1.3,16c-0.7,0-1.1-0.3-1.2-0.8c-0.3-0.8,0.5-1.3,0.8-1.5c0.6-0.4,0.9-0.7,1-1c0-0.2-0.1-0.4-0.3-0.7c0,0,0-0.1-0.1-0.1 C0.5,10.6,0,9,0,7.4C0,3.3,3.4,0,7.5,0C11.6,0,15,3.3,15,7.4s-3.4,7.4-7.5,7.4c-0.5,0-1-0.1-1.5-0.2C3.4,15.9,1.5,16,1.5,16 C1.4,16,1.4,16,1.3,16z M3.3,10.9c0.5,0.7,0.7,1.5,0.6,2.2c0,0.1-0.1,0.3-0.1,0.4c0.5-0.2,1-0.4,1.6-0.7c0.2-0.1,0.4-0.2,0.6-0.1 c0,0,0.1,0,0.1,0c0.4,0.1,0.9,0.2,1.4,0.2c3,0,5.5-2.4,5.5-5.4S10.5,2,7.5,2C4.5,2,2,4.4,2,7.4c0,1.2,0.4,2.4,1.2,3.3 C3.2,10.8,3.3,10.8,3.3,10.9z"></path></svg><span>Chat</span>';
+
+        btn.className =
+          "b br-pill pointer bg-base--inverted white-90 flex items-center justify-center";
+
+        btn.addEventListener("click", function () {
+          localStorage.setItem("zdChatOpen", true);
+          addZDSnippet();
+        });
+
+        document.body.appendChild(btn);
+      }
+    });
+
+    window.addEventListener(
```

Comment:
That event listener can added in the specific element instead of window?

Reply:
Nope, it's a document event

https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event

Reply:
It's not necessaire right now, it's just a nice to have in case zendesk changes their localStorage item names.

Reply:
show!